product_attribute_custom_value_variant/models/sale_order_line.py

```diff
@@ -1,60 +1,87 @@
 # Copyright 2025 Simone Rubino - Aion Tech
 # License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
-
 from odoo import api, models
 
+from odoo.addons.product.models.product_attribute_custom_value import (
+    ProductAttributeCustomValue,
+)
+from odoo.addons.product.models.product_attribute_value import ProductAttributeValue
+
 
 class SaleOrderLine(models.Model):
     _inherit = "sale.order.line"
 
+    def _get_new_and_unlink_custom_attribute(
+        self,
+    ) -> tuple[ProductAttributeValue, ProductAttributeCustomValue]:
```

Comment:
This is interesting, are we using types now in Odoo? Maybe I'm not updated with the latest trends, do you have any reference?

Reply:
Typing is great to auto document the code (it's Odoo here, but it's valid for other projects) as it's standard python and does not harm the code.